js/ui/windowManager.js

```diff
@@ -1010,8 +1126,18 @@ const ShellWindowManagerAnimatableSurface = new Lang.Class({
     },
 
     vfunc_attach_effect: function(event, effect) {
-        if ((_ALLOWED_ANIMATIONS_FOR_EVENTS[event] || []).indexOf(effect.name) == -1) {
-            throw new Error(`Effect ${effect.name} can't be used on event ${event}`);
+        let effects = _ALLOWED_ANIMATIONS_FOR_EVENTS[event];
+
+        if (!effects) {
+            throw new GLib.Error(AnimationsDbus.error_quark(),
+                                 AnimationsDbus.Error.UNSUPPORTED_EVENT_FOR_ANIMATION_SURFACE,
+                                 `Surface does not support event ${event}`);
+        }
+
+        if ((effects || []).indexOf(effect.name) == -1) {
```

Comment:
Maybe ensure the effects are `[]` on their assignment?

Reply:
Fixed